### ISSUE TITLE
Incorrect "MachineCodeModelNotSupported" when build command has no zig files.

### ISSUE THREAD

**Issue by rgreenblatt** [ziglang/zig#7900]
The following command fails with `error: unable to create compilation: MachineCodeModelNotSupported`
```
zig build-lib -mcmodel=large temp.c
```
However,
```
zig build-lib -mcmodel=large file.c file.zig
```
doesn't fail.

**Comment by Matthew-Amos** [comment]
Looks like this is because the compiler does not default to using LLVM with the C file. Using the following command will alleviate this error (add the -fLLVM flag):

```
zig build-lib -mcmodel=large -fLLVM temp.c
```

I believe this behavior originates from [Compilation.zig#L580](https://github.com/ziglang/zig/blob/master/src/Compilation.zig#L580). Full disclosure that this is the first time I'm looking at any of this, but it seems to me that it's the first condition in that block that gets hit since `build_crt_file` sets `root_pkg` to `null`, thus disabling LLVM by default with the current design.

**Comment by rgreenblatt** [comment]
> ```
> zig build-lib -mcmodel=large -fLLVM temp.c
> ```

I tried this when I first ran into this issue, but it segfaults (at least on my computer on 0.8.0-dev.1082+bf76501b5). I thought that this was a known issue and `-fLLVM` isn't supported.
Perhaps `-fLLVM` is supposed to work?

**Comment by g-w1** [comment]
Its not, its the experimental llvm stage2 backend.

**Comment by andrewrk** [comment]
Confirmed that it is working now. Looks like I fixed it as a side effect of d11f42c2b24e80388e1ea648ee97fae612fd76a4.
